### PR TITLE
[IMP] mail: add active/archive for restrict deletion

### PR DESCRIPTION
**Pull request by nsa-odoo**
Description of the issue/feature this PR : 
 task : https://www.odoo.com/web#id=36122&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
 pad : https://pad.odoo.com/p/r.d548146a1756aa88e6880b99f0f9023e
Current behavior before PR: 

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_activity.py [hunk 1]
```diff
@@ -66,7 +67,7 @@ def default_get(self, fields):
     # activity
     activity_type_id = fields.Many2one(
         'mail.activity.type', 'Activity',
-        domain="['|', ('res_model_id', '=', False), ('res_model_id', '=', res_model_id)]")
+        domain="['|', ('res_model_id', '=', False), ('res_model_id', '=', res_model_id)]" ,ondelete ='restrict')
```
Comment (mreficent) [inline addons/mail/models/mail_activity.py]: `"_,` → `",_`

File: addons/mail/models/mail_activity.py [hunk 2]
```diff
@@ -39,6 +40,12 @@ class MailActivityType(models.Model):
         help='Categories may trigger specific behavior like opening calendar view')
 
 
+    @api.multi
+    def unlink(self):
+        if self.env['mail.activity'].search([('activity_type_id', 'in', self.ids)], limit=1):
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_activity.py]: You can use a search_count instead

File: addons/mail/views/mail_activity_views.xml [hunk 3]
```diff
@@ -27,6 +32,17 @@
         </field>
     </record>
 
+    <record id="mail_activity_type_view_search" model="ir.ui.view">
+            <field name="name">mail.activity.type.search</field>
+            <field name="model">mail.activity.type</field>
+            <field name="arch" type="xml">
+                <search string="Activities search">
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_activity_views.xml]: No need to add the search, Activities is sufficient

File: addons/mail/views/mail_activity_views.xml [hunk 4]
```diff
@@ -27,6 +32,17 @@
         </field>
     </record>
 
+    <record id="mail_activity_type_view_search" model="ir.ui.view">
+            <field name="name">mail.activity.type.search</field>
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_activity_views.xml]: Guidelines :) !

File: addons/mail/views/mail_activity_views.xml [hunk 5]
```diff
@@ -27,6 +32,17 @@
...
+            <field name="model">mail.activity.type</field>
+            <field name="arch" type="xml">
+                <search string="Activities search">
+                    <field name="name"/>
+                    <filter string="Archived" domain="[('active', '=', False)]" name="archive"/>
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_activity_views.xml]: Don't you think name and string should match more ?

File: addons/mail/models/mail_activity.py [hunk 6]
```diff
@@ -39,6 +40,12 @@ class MailActivityType(models.Model):
         help='Categories may trigger specific behavior like opening calendar view')
 
 
+    @api.multi
+    def unlink(self):
+        if self.env['mail.activity'].search_count([('activity_type_id', 'in', self.ids)]):
+            raise UserError(_("This activity type is already used in scheduled activites. If you no longer want to use it, you can archive it."))
+        return super(MailActivityType, self).unlink()
+
```
Comment (rim-odoo) [inline addons/mail/models/mail_activity.py]: @tde-banana-odoo Here we are re-implementing an `ondelete='restrict'` for the sake of change the error message... not sure it is worthy

Comment (tde-banana-odoo) [reply addons/mail/models/mail_activity.py]: Seems right.

**Comment by pga-odoo** [comment]
LGTM

**Pull request merged**